File: v22_chain_of_responsibility/java/LoggerDemo1.java
package v22_chain_of_responsibility.java;

class Logger {
    public static final int ERROR = 1;
    public static final int DEBUG = 2;
    public static final int INFO = 3;
    private int level;

    public Logger(int level) {
        this.level = level;
    }

    public void logMessage(int logLevel, String message) {
        if (logLevel == ERROR && logLevel <= level) {
            System.out.println("Error: " + message);
        } else if (logLevel == DEBUG && logLevel <= level) {
            System.out.println("Debug: " + message);
        } else if (logLevel == INFO && logLevel <= level) {
            System.out.println("Info: " + message);
        }
    }
}

public class LoggerDemo1 {
    public static void main(String[] args) {
        Logger logger = new Logger(Logger.INFO);

        logger.logMessage(Logger.ERROR, "This is an error message");
        logger.logMessage(Logger.DEBUG, "This is a debug message");
        logger.logMessage(Logger.INFO, "This is an info message");
    }
}

File: v22_chain_of_responsibility/python/logger_demo1.py
class Logger:
    ERROR = 1
    DEBUG = 2
    INFO = 3

    def __init__(self, level):
        self.level = level

    def log_message(self, log_level, message):
        if log_level == self.ERROR and log_level <= self.level:
            print("Error: " + message)
        elif log_level == self.DEBUG and log_level <= self.level:
            print("Debug: " + message)
        elif log_level == self.INFO and log_level <= self.level:
            print("Info: " + message)


if __name__ == "__main__":
    logger = Logger(Logger.INFO)

    logger.log_message(Logger.ERROR, "This is an error message")
    logger.log_message(Logger.DEBUG, "This is a debug message")
    logger.log_message(Logger.INFO, "This is an info message")


